### DISCUSSION TITLE
mypy `_PlotMethods` attributes

### DISCUSSION THREAD

**Discussion by zmoon** [pydata/xarray#6947] [Q&A] [answered]
With current xarray release I am now getting errors from mypy like this one
```
error: "_PlotMethods" has no attribute "contourf"
```
What is recommended way to resolve (or get around) this?

**Comment by headtr1ck** [comment, marked as answer]
That is a bit unfortunate.
Now the `.plot` attribute is correctly typed as the `_PlotMethods` accessor which however is not typed yet.
The problem with this accessor is that many functions are added via hook methods which does not work well with typing.

For now your only possibility is to use `#type: ignore`.

For the next release we should either type the accessors or add an Any type as an escape hatch to avoid these false positives.

I have opened #6949 for this problem.

**Reply by zmoon** [reply]
I used `# type: ignore[attr-defined]` to be more specific. Thanks @headtr1ck for the explanation and doing the issue.
